### PR TITLE
generate systemd: quote arguments with whitespace

### PR DESCRIPTION
**Pull request by vrothberg**
Make sure that arguments with whitespace are properly quoted so they are
interpreted as one (and not multiple ones) by systemd.

Now `-e tz="america/new york"` will be generated as `-e "tz=america/new york"`.
The quotes are moving but the argument is still correct.

Fixes: #7285
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
@mheon a late candidate for 2.0.

@edsantiago, I need a head nod from you before we can merge. Can you think of a case where the changes could lead to a regression?

**Comment by rhatdan** [comment]
LGTM

File: pkg/systemd/generate/common.go [hunk 1]
```diff
@@ -53,3 +55,18 @@ func filterPodFlags(command []string) []string {
...
+
+// regexWhitespace matches string that contain at least one whitespace.
+var regexWhitespace = regexp.MustCompile(`^.*\s.*$`)
+
+// quoteArguments makes sure that all arguments with at least one whitespace
+// are quoted to make sure those are interpreted as one argument instead of
+// multiple ones.
+func quoteArguments(command []string) []string {
+	for i := range command {
+		if regexWhitespace.MatchString(command[i]) {
```
Comment (edsantiago) [inline pkg/systemd/generate/common.go:67]: regex seems like such overkill. Did you consider `string.ContainsAny` with `"<space><tab>"`?

File: pkg/systemd/generate/pods_test.go [hunk 2]
```diff
@@ -118,7 +118,7 @@ WantedBy=multi-user.target default.target`
 				StopTimeout:      10,
 				PodmanVersion:    "CI",
 				RequiredServices: []string{"container-1", "container-2"},
-				CreateCommand:    []string{"podman", "pod", "create", "--name", "foo"},
+				CreateCommand:    []string{"podman", "pod", "create", "--name", "foo", "bar=arg with space"},
```
Comment (edsantiago) [inline pkg/systemd/generate/pods_test.go:121]: I can't quickly figure out whether `strconv.Quote()` handles quotes themselves. Could you change this to `"bar=arg \"with space\""` and have the test itself guarantee that quotes are preserved?

File: pkg/systemd/generate/common.go (hunk 1 above)
Comment (vrothberg) [reply pkg/systemd/generate/common.go:67]: Good idea, I'll change!

File: pkg/systemd/generate/pods_test.go (hunk 2 above)
Comment (vrothberg) [reply pkg/systemd/generate/pods_test.go:121]: Quotes are preserved. I added additional unit tests. Cool with these?
https://github.com/containers/podman/pull/7350/files#diff-1b78823cd7c48fb19dc371e2eb287729R32-R55

Comment (edsantiago) [reply pkg/systemd/generate/pods_test.go:121]: Yes! Nice!

Comment (vrothberg) [reply pkg/systemd/generate/pods_test.go:121]: Great, thanks!

Can you /lgtm /hold?

**Comment by mheon** [comment]
Nice catch, LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
and I think #7195 is biting this too

**Comment by edsantiago** [comment]
No, it's the CI hang. @vrothberg you need to rebase on master, I'm sorry.

**Comment by vrothberg** [comment]
merge me :)

**Pull request merged**